shell/common/shell.cc

```diff
@@ -321,33 +271,34 @@ Shell::~Shell() {
   PersistentCache::GetCacheForProcess()->RemoveWorkerTaskRunner(
       task_runners_.GetIOTaskRunner());
 
-  vm_->GetServiceProtocol()->RemoveHandler(this);
-
   fml::AutoResetWaitableEvent ui_latch, gpu_latch, platform_latch, io_latch;
 
   fml::TaskRunner::RunNowOrPostTask(
       task_runners_.GetUITaskRunner(),
-      fml::MakeCopyable([engine = std::move(engine_), &ui_latch]() mutable {
-        engine.reset();
+      fml::MakeCopyable([shell = GetShell(), &ui_latch]() mutable {
+        shell->vm_->GetServiceProtocol()->RemoveHandler(shell.get());
```

Comment:
This seems very dangerous.  We're almost definitely going to see situations where the shell pointer is being used on the wrong thread and after the destruction of the object.

Reply:
Could we use [this] directly?

Reply:
I'm not really sure how that helps anything.  We're still be using a pointer to the shell at a point where its destructor has finished.